### DISCUSSION TITLE
集成keycloak页面回跳回来之后路由没有匹配到对应的页面

### DISCUSSION THREAD

**Discussion by KingDracula** [umijs/umi#9728] [Q&A]
在antdesign Pro react 里集成keycloak  页面初始化时进行登录校验，校验完会跳回页面刷新时的页面，但是没有匹配到对应的路由上重新渲染页面，直接显示404 页面，但是重新点击一下首页按钮后就可以正常显示了
请教一下这个问题应该怎么解决？

export default [
  {
    path: '/',
    component: '../layouts/SecurityLayout',

    routes: [
      {
        path: '/',
        component: '../layouts/BlankLayout',
        routes: [
          {
            path: '/',
            component: '../layouts/BasicLayout',

            routes: [
              // API Repository
              {
                path: '/api-repo',
                redirect: '/api-repo/list',
              },
              {
                name: 'api-list',
                path: '/api-repo/list',
                component: './api-repo/list',
              },
              {
                name: 'api-detail',
                path: '/api-repo/detail/:id',
                component: './api-repo/detail',
              },
              {
                exact: true,
                name: 'create-rest-api',
                icon: 'smile',
                path: '/api-repo/create/:id',
                component: './api-repo/create',
              },
              {
                exact: true,
                name: 'edit-rest-api',
                icon: 'smile',
                path: '/api-repo/edit/:id',
                component: './api-repo/create',
              },
              {
                path: '/',
                redirect: '/api-repo/list',
              },
              {
                component: './404',
              },
            ],
          },
          {
            component: './404',
          },
        ],
      },
    ],
  },
  {
    component: './404',
  },
];

**Comment by fz6m** [comment]
给一个最小复现。
